ft-sdk/src/layout.rs

```diff
@@ -88,8 +104,37 @@ pub trait Layout {
         Ok(a2r(o))
     }
 
+    fn action_with_log<A>(r: http::Request<bytes::Bytes>) -> http::Response<bytes::Bytes>
+        where
+            A: ActionWithLog<Self, Self::Error>,
+            Self: Sized,
+    {
+        match Self::_action_with_log::<A>(r) {
+            Ok(r) => r,
+            Err(e) => Self::render_error(e),
+        }
+    }
+
+    fn _action_with_log<A>(
+        r: http::Request<bytes::Bytes>,
+    ) -> Result<http::Response<bytes::Bytes>, Self::Error>
+        where
+            A: ActionWithLog<Self, Self::Error>,
+            Self: Sized,
+    {
+        let start_time = ft_sys::now();
+        let in_ = ft_sdk::In::from_request(r)?;
+        let mut l = Self::from_in(in_, RequestType::Action)?;
+        let a = A::validate(&mut l)?;
+        let o = a.action_with_log(&mut l)?;
```

Comment:
This line must internally create the log. Since log need global data, and that can be handled, in_, which is provided by end user has to store the partial log data.